eclass/multiprocessing.eclass

```diff
@@ -64,17 +64,36 @@ get_nproc() {
 	fi
 }
 
+# @FUNCTION: _get_all_makeopts
+# @INTERNAL
+# @DESCRIPTION:
+# Returns ${MAKEOPTS} ${GNUMAKEFLAGS} ${MAKEFLAGS}.
+_get_all_makeopts() {
+	echo "${MAKEOPTS} ${GNUMAKEFLAGS} ${MAKEFLAGS}"
+}
+
+# @FUNCTION: get_makeopts_jobs
+# @USAGE: [default-jobs]
+# @DESCRIPTION:
+# Return the number of jobs extracted from the make options (MAKEOPTS,
+# GNUMAKEFLAGS, MAKEFLAGS). If the make options do not specify a number,
+# then either the provided default is returned, or 1.
+get_makeopts_jobs() {
+	local makeopts="$(_get_all_makeopts)"
+	echo $(makeopts_jobs ${makeopts} ${1:-1})
```

Comment:
I think makeopts needs to be quoted here.

Probably best to quote the second argument as well, in case somebody passes some whitespace in error.

```suggestion
	echo $(makeopts_jobs "${makeopts}" "${1:-1}")
```